File: mobile/lib/data/services/input_mapping.dart
import 'package:shared_preferences/shared_preferences.dart';

/// How dashboard buttons are interpreted by the desktop.
enum InputMapping {
  /// Dashboard controls become simulated keyboard presses (ETS2 defaults).
  keyboard('keyboard'),

  /// Dashboard controls become virtual-controller button presses.
  gamepad('gamepad');

  const InputMapping(this.wireValue);

  /// Value used in the `mapping` frame, shared with the desktop.
  final String wireValue;

  static const String prefsKey = 'wheeldeck.input_mapping';

  static InputMapping fromWireValue(String? value) => InputMapping.values
      .firstWhere((m) => m.wireValue == value, orElse: () => keyboard);

  /// Loads the persisted mapping, defaulting to keyboard.
  static Future<InputMapping> load() async {
    final prefs = await SharedPreferences.getInstance();
    return fromWireValue(prefs.getString(prefsKey));
  }

  /// Persists the mapping.
  Future<void> save() async {
    final prefs = await SharedPreferences.getInstance();
    await prefs.setString(prefsKey, wireValue);
  }
}

File: android/app/src/main/kotlin/dev/fazrigading/wheeldeck/data/services/InputMapping.kt
package dev.fazrigading.wheeldeck.data.services

import dev.fazrigading.wheeldeck.domain.models.MappingMode

/// How dashboard buttons are interpreted by the desktop.
/// Persistence lands with the settings store (plan Task 13).
enum class InputMapping(val wireValue: String) {
    /** Dashboard controls become simulated keyboard presses (ETS2 defaults). */
    Keyboard("keyboard"),

    /** Dashboard controls become virtual-controller button presses. */
    Gamepad("gamepad");

    companion object {
        val fallback = Gamepad

        fun fromWireValue(value: String?): InputMapping =
            entries.firstOrNull { it.wireValue == value } ?: fallback

        fun fromMappingMode(mode: MappingMode): InputMapping = valueOf(mode.name)
    }
}
